cpp-terminal/terminal.h

```diff
@@ -403,7 +401,11 @@ class Terminal: public BaseTerminal {
         i = 0;
         while (i < sizeof(buf) - 1 - 5) {
             if (buf[i] == '\x1b' && buf[i+1] == '[') {
+#ifdef _WIN32
+                if (sscanf_s(&buf[i+2], "%d;%d", &rows, &cols) == 2) {
+#else
                 if (sscanf(&buf[i+2], "%d;%d", &rows, &cols) == 2) {
+#endif
```

Comment:
The way `cpp-terminal` is designed is that the `terminal.h` file is strictly platform independent, and thus does not have any kind of platform specific `ifdef` statements. The file `terminal_base.h` is platform dependent. As such, this should be moved there in some way. Do you think you could please do that?

Why is the `sscanf_s` needed? I thought Windows tests are passing in master with the `sscanf` function.

Reply:
Yes, i will try to move that. The thing with sscanf is, that MSVC compiles it just fine, but throws a warning. That's, because microsoft provides their own implementation (sscanf_s) which is probably safer in terms of conversion errors. Linux don't offers that function, but clang-tidy notes:
```
'sscanf' used to convert a string to an integer value, but function will not report conversion errors; consider using 'strtol' instead
```
So i think, it would be a great thing to implement sscanf_s on windows and strtol on linux (even though strtol will require more effort because of different arguments).

Reply:
What is the C++ way to do what I am doing there? Using the regex standard library?

Reply:
I'm not sure, if there is a "C++ way of string conversion", the c++ reference [(here)](http://www.cplusplus.com/reference/cstdio/sscanf/) lists it inside both <stdio.h> and <cstdio>, so it's actually inside the C++ standard. Ofcourse you could say, that using the stdio library is bad, because it just offers some C functions inside C++ and is in fact not a "real" part of the C++ standard namespace. I have thought about maybe using std::stringstream. I will investigate both solutions and review my findings later.
Thank you.

Reply:
The easiest to get this PR merged is to simply provide our own `sscanf` method in `terminal_base.h` that dispatches to `sscanf_s` on Windows using `ifdef`. Then we simply call it from `terminal.h`.

Later, we can figure out if there is a better way.